src/components/LocationAutocomplete/LocationAutocompleteRenderer/LocationAutocompleteRenderer.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import { Autosuggest } from '../../Autosuggest';
+import useDebounce from '../../../hooks/useDebounce';
+import { ListItem, MarkedText } from '../../../index';
+import POWERED_BY_GOOGLE_ON_WHITE from '../../../images/powered_by_google_on_white.png';
+import styles from '../LocationAutocomplete.scss';
+
+const { elem } = bem({
+    name: 'LocationAutocomplete',
+    classnames: styles,
+});
+
+const DEBOUNCE_DELAY = 350;
+
+const LocationAutocomplete = props => {
+    const [inputValue, setInputValue] = React.useState('');
+    const [suggestionsList, setSuggestionsList] = React.useState([]);
+    const [isLoading, setIsLoading] = React.useState(false);
+
+    const {
+        onSelectionChange,
+        inputPlaceholder,
+        noSuggestionsPlaceholder,
+        country,
+        placeTypes,
+        ...rest
+    } = props;
+
+    // Suggestion functions
+    const resetSuggestionsList = () => setSuggestionsList([]);
+    const suggestionToString = suggestion => (suggestion ? suggestion.description : '');
+
+    const debouncedInputValue = useDebounce(inputValue, DEBOUNCE_DELAY);
+
+    React.useEffect(() => {
+        if (debouncedInputValue) {
+            const service = new window.google.maps.places.AutocompleteService();
+
+            service.getPlacePredictions(
+                {
+                    input: debouncedInputValue,
+                    types: placeTypes,
+                    componentRestrictions: { country },
+                },
+                predictions => {
+                    setSuggestionsList(predictions);
+                    setIsLoading(false);
+                }
+            );
+        } else {
+            resetSuggestionsList();
+        }
+    }, [country, debouncedInputValue, placeTypes]);
+
+    const handleInputValueChange = value => {
+        if (value) {
+            setIsLoading(true);
+            setInputValue(value);
+        } else {
+            setIsLoading(false);
+            resetSuggestionsList();
+        }
+    };
+
+    // eslint-disable-next-line react/display-name
+    const renderListPoweredByGoogle = ({ listInputValue, getItemProps, highlightedIndex }) => (
```

Comment:
A little bit strange name and I'd prefer to split up this method on one render list other insert img

Reply:
the list is rendered by autosuggest, this method overwrites the default behaviour of autosuggest. I think this is the simplest way to get everything into the proper place in the dom structure.